horovod/mxnet/mpi_ops.cc

```diff
@@ -241,17 +230,17 @@ extern "C" int horovod_mxnet_allgather_async(
 }
 
 extern "C" int horovod_mxnet_broadcast_async(
-    NDArray* tensor, NDArray* output, int root_rank, char* name, Callback cb) {
+    NDArray* tensor, int root_rank, char* name, Callback cb) {
 #if HOROVOD_GPU_ALLGATHER == 'M'   
   if (tensor->ctx().dev_mask() == gpu::kDevMask &&
       output->ctx().dev_mask() == gpu::kDevMask)
-    return DoBroadcast(tensor, output, root_rank, name, cb);
+    return DoBroadcast(tensor, root_rank, name, cb);
   else
 #endif
 #if HAVE_CUDA
```

Comment:
Since you are now doing the GPU to CPU copy in the MXNet side, I believe you can get rid of all the #if's and if's by just calling `DoBroadcast` instead of `DoBroadcastCudaOnCPU`